apps/meteor/app/api/server/v1/teams.ts

```diff
@@ -576,6 +576,15 @@ API.v1.addRoute(
 				return API.v1.failure('Team not found');
 			}
 
+			const canViewInfo =
+				(await Team.getMember(teamInfo._id, this.userId)) ||
+				(await Subscriptions.findOneByRoomIdAndUserId(teamInfo.roomId, this.userId)) ||
+				hasPermission(this.userId, 'view-all-teams');
```

Comment:
I think a better option here would be to use the `canAccessRoom` method instead of checking if the user is a member of the team and searching for the subscription directly. By doing so, we'd no longer need the `getMember` method.

Reply:
updated!